module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -846,18 +855,29 @@ namespace app
 
     bool ApplicationCommon::tryShowPopup()
     {
+        using namespace gui::popup;
+
         auto request = windowsPopupQueue->popRequest(getPopupFilter());
-        if (request) {
-            const auto popup = magic_enum::enum_name(request->getPopupParams().getPopupId()).data();
-            LOG_DEBUG("handling popup: %s", popup);
-            /// request handle actually switches window to popup window
-            auto retval = request->handle();
-            if (not retval) {
-                LOG_ERROR("Popup %s handling failure, please check registered blueprint!", popup);
-            }
-            return retval;
+        if (!request) {
+            return false;
         }
-        return false;
+
+        auto const id = request->getPopupParams().getPopupId();
+
+        if (id == ID::SimInfo) {
+            // MOS-350: silently get rid of the first occurence of SimLock pop-up and all subsequent windows
+            windowsStack().drop(resolveWindowName(ID::SimLock));
```

Comment:
does the SimLock pop-up appear always twice?

Reply:
Maybe you misread the intention of the comment: it's not that the pop-up occurs twice, it's that `drop()` - by its nature - searches from the beginning and stop at the first occurrence, no matter how many there are yet.

Reply:
alright, makes sense then ;)